### PR TITLE
Generate graphql SDL file from daemon

### PR DESCRIPTION
**Pull request by Schmavery**
This replaces the `schema.graphql` file with one generated using the daemon graphql server. This replaces the hand-written faker schema but still allows us to run the faker and (near-future) graphdoc generation without needing to have the daemon running.

Since we're generating the schema now, we don't get the cute `@fake` annotations, so I had to tweak some of our decoders to inject fake data themselves when faker is running.

I realized while testing the wallet with the new schema that graphql variables don't seem to get typechecked properly which is fun. I updated all the spots I could find to use the new scalars. 

The ChildProcess.spawn bindings changes were left over from a workaround I tried on the way to getting this working, but it was an improvement so I left it in.

Not included in this PR:
- Actual Graphdoc generation
- All the weird perl/sed scripts I tried to patch the autogenerated sdl file with -- note to self: if you end up doing process substitution in child_process.spawn call you should probably reexamine your life decisions

### CODE REVIEW

File: frontend/wallet/src/render/Apollo.re [hunk 1]
```diff
@@ -33,14 +33,43 @@ let client = {
 };
 
 module Decoders = {
-  let int64 = pk =>
-    Js.Json.decodeString(pk) |> Option.getExn |> Int64.of_string;
+  [@bs.val] [@bs.scope "window"] external isFaker: bool = "";
```
Comment (o1pranay) [inline frontend/wallet/src/render/Apollo.re]: qq - I get that this is telling bucklescript to get `isFaker` from window, but don't understand why you have to add an empty string literal as the body?  Is that just convention?

File: frontend/wallet/src/render/views/footer/Footer.re [hunk 2]
```diff
@@ -131,10 +131,10 @@ let make = () => {
                        ) => {
                          let variables =
                            SendPayment.make(
-                             ~from=PublicKey.toString(from),
-                             ~to_=PublicKey.toString(to_),
-                             ~amount,
-                             ~fee,
+                             ~from=Apollo.Encoders.publicKey(from),
+                             ~to_=Apollo.Encoders.publicKey(to_),
+                             ~amount=Js.Json.string(amount),
```
Comment (o1pranay) [inline frontend/wallet/src/render/views/footer/Footer.re]: why not use `Apollo.Encoders.int64` here?

File: frontend/wallet/src/render/Apollo.re [hunk 3]
```diff
@@ -33,14 +33,43 @@ let client = {
...
+
+  let int64 = pk => {
+    let s = Option.getExn(Js.Json.decodeString(pk));
```
Comment (o1pranay) [inline frontend/wallet/src/render/Apollo.re]: im starting to like the pipe syntax more and more everyday lol

File: frontend/wallet/src/render/Apollo.re (hunk 1 above)
Comment (Schmavery) [reply frontend/wallet/src/render/Apollo.re]: It's short for `external isFaker: bool = "isFaker";`. The first `isFaker` is the name of the ocaml/reason function you're defining and the second `"isFaker"` is the name of the JS (or if you're writing a native external, C) function you're binding to.

File: frontend/wallet/src/render/views/footer/Footer.re (hunk 2 above)
Comment (Schmavery) [reply frontend/wallet/src/render/views/footer/Footer.re]: Apollo.Encoders.int64 converts from int64 to Js.Json.t, here amount is a string

File: frontend/wallet/src/render/Apollo.re (hunk 1 above)
Comment (o1pranay) [reply frontend/wallet/src/render/Apollo.re]: got it, thx

File: frontend/wallet/src/render/views/footer/Footer.re (hunk 2 above)
Comment (o1pranay) [reply frontend/wallet/src/render/views/footer/Footer.re]: oh duh, my b. It's weird though in the `validate` function in SendButton, we convert the fee & amount to int64 and then return strings, while we convert the toStr into a PublicKey.t and return that - should be consistent here

Comment (Schmavery) [reply frontend/wallet/src/render/views/footer/Footer.re]: fair enough

Comment (Schmavery) [reply frontend/wallet/src/render/views/footer/Footer.re]: This is taking too long, I'm going to merge and hopefully rework this stuff later

**Pull request merged**